gsitemap.php

```diff
@@ -488,6 +494,131 @@ protected function getCategoryLink(&$link_sitemap, $lang, &$index, &$i, $id_cate
+			'SELECT m.`id_manufacturer` FROM `'._DB_PREFIX_.'manufacturer` m
+			INNER JOIN `'._DB_PREFIX_.'manufacturer_lang` ml on m.`id_manufacturer` = ml.`id_manufacturer`'.
+			($this->tableColumnExists(_DB_PREFIX_.'manufacturer_shop') ? ' INNER JOIN `'._DB_PREFIX_.'manufacturer_shop` ms ON m.`id_manufacturer` = ms.`id_manufacturer` ' : '').
+			' WHERE m.`active` = 1  AND m.`id_manufacturer` >= '.(int)$id_manufacturer.
+			($this->tableColumnExists(_DB_PREFIX_.'manufacturer_shop') ? ' AND ms.`id_shop` = '.(int)$this->context->shop->id : '').
+			' AND ml.`id_lang` = '.(int)$lang['id_lang'].
+			' ORDER BY m.`id_manufacturer` ASC'
+		);
+		foreach ($manufacturers_id as $manufacturer_id)
+		{
+			$manufacturer = new Manufacturer((int)$manufacturer_id['id_manufacturer'], $lang['id_lang']);
+			$url = $link->getManufacturerLink($manufacturer, $manufacturer->link_rewrite, $lang['id_lang']);
+            
+            $image_link = $this->context->link->getManufacturerImageLink((int) $manufacturer->id, ImageType::getFormattedName('manufacturer'));
+			$image_link = (!in_array(rtrim(Context::getContext()->shop->virtual_uri, '/'), explode('/', $image_link))) ? str_replace(array(
+                'https',
+                Context::getContext()->shop->domain . Context::getContext()->shop->physical_uri
+            ), array(
+                'http',
+                Context::getContext()->shop->domain . Context::getContext()->shop->physical_uri . Context::getContext()->shop->virtual_uri
+            ), $image_link) : $image_link;
+
+			$file_headers = (Configuration::get('GSITEMAP_CHECK_IMAGE_FILE')) ? @get_headers($image_link) : true;
+			$manifacturer_image = array();
+			if ($file_headers[0] != 'HTTP/1.1 404 Not Found' || $file_headers === true)
+				$manifacturer_image = array(
+					'title_img' => htmlspecialchars(strip_tags($manufacturer->name)),
+					'caption' => htmlspecialchars(strip_tags($manufacturer->short_description)),
+					'link' => $image_link
+				);
+			if (!$this->addLinkToSitemap(
+				$link_sitemap, array(
+					'type' => 'manufacturer',
+					'page' => 'manufacturer',
+					'lastmod' => $manufacturer->date_upd,
+					'link' => $url,
+					'image' => $manifacturer_image
+				), $lang['iso_code'], $index, $i, $manufacturer_id['id_manufacturer']
+			))
+				return false;
+		}
+
+		return true;
+	}
+
+    /**
+	 * @param array  $link_sitemap contain all the links for the Google Sitemap file to be generated
+	 * @param string $lang         language of link to add
+	 * @param int    $index        index of the current Google Sitemap file
+	 * @param int    $i            count of elements added to sitemap main array
+	 * @param int    $id_supplier  supplier object identifier
+	 *
+	 * @return bool
+	 */
+	protected function getSupplierLink(&$link_sitemap, $lang, &$index, &$i, $id_supplier = 0)
+	{
+		$link = new Link();
+		if (method_exists('ShopUrl', 'resetMainDomainCache')) {
+            ShopUrl::resetMainDomainCache();
```

Comment:
Could you please you use spaces everywhere, indentation is broken.

Reply:
Done